src/main/java/seedu/tr4cker/model/util/NaturalDateUtil.java

```diff
@@ -0,0 +1,86 @@
+package seedu.tr4cker.model.util;
+
+import static java.time.DayOfWeek.FRIDAY;
+import static java.time.DayOfWeek.MONDAY;
+import static java.time.DayOfWeek.SATURDAY;
+import static java.time.DayOfWeek.SUNDAY;
+import static java.time.DayOfWeek.THURSDAY;
+import static java.time.DayOfWeek.TUESDAY;
+import static java.time.DayOfWeek.WEDNESDAY;
+import static java.time.temporal.TemporalAdjusters.next;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import seedu.tr4cker.logic.parser.exceptions.ParseException;
+
+public class NaturalDateUtil {
+
+    public static final String MESSAGE_DAY = "Please enter a valid day of the week in full, e.g. Saturday";
+    public static final String MESSAGE_TIME = "Please enter a valid time in 24Hr format";
+    public static final String VALIDATION_REGEX = "[a-zA-Z]{5,}";
+    public static final String VALIDATION_REGEX_TIME = "[a-zA-Z]{5,} \\d{4}";
+
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd-MMM-yyyy");
+    private static final Map<String, DayOfWeek> DAYS_OF_WEEK;
+
+    static {
+        DAYS_OF_WEEK = new HashMap<>();
+        DAYS_OF_WEEK.put("MONDAY", MONDAY);
+        DAYS_OF_WEEK.put("TUESDAY", TUESDAY);
+        DAYS_OF_WEEK.put("WEDNESDAY", WEDNESDAY);
+        DAYS_OF_WEEK.put("THURSDAY", THURSDAY);
+        DAYS_OF_WEEK.put("FRIDAY", FRIDAY);
+        DAYS_OF_WEEK.put("SATURDAY", SATURDAY);
+        DAYS_OF_WEEK.put("SUNDAY", SUNDAY);
+        DAYS_OF_WEEK.put("TODAY", null);
+    }
+
+    /**
+     * Returns true if a given string is a Natural Deadline.
+     */
+    public static boolean isNaturalDeadline(String test) {
+        return test.matches(VALIDATION_REGEX) || test.matches(VALIDATION_REGEX_TIME);
+    }
+
+    /**
+     * Converts a natural date-time input into a formatted DateTime String.
+     *
+     * @param naturalDateTime natural date-time input
+     * @return formatted DateTime String
+     * @throws ParseException when the date or time is in the wrong format
+     */
+    public static String convertToDateTime(String naturalDateTime) throws ParseException {
+        assert isNaturalDeadline(naturalDateTime);
+        String dateTime;
+        String [] deadlineTokens = naturalDateTime.split(" ");
+        int unitsOfDeadline = deadlineTokens.length;
+        assert unitsOfDeadline == 1 || unitsOfDeadline == 2;
+
+        String dayOfDeadline = deadlineTokens[0].toUpperCase();
+
+        if (!DAYS_OF_WEEK.containsKey(dayOfDeadline)) {
+            throw new ParseException(MESSAGE_DAY);
+        } else if (dayOfDeadline.equals("TODAY")) {
```

Comment:
if ur checking separately for TODAY, perhaps TODAY does not need to be inside the hashmap?